app/controllers/atmosphere/api/v1/appliance_types_controller.rb

```diff
@@ -93,6 +78,14 @@ def endpoint_payload
 
         private
 
+        def check_for_conflict!(appl)
+          vm = appl.virtual_machines.first
+          if vm && vm.state.saving?
+            raise Atmosphere::Conflict,
+                  'It is not allowed to save application twice'
```

Comment:
Może "This service is already being saved. Please wait for this operation to conclude before attempting to save the service again."